data_structures/queue/queue.c

```diff
@@ -48,7 +51,8 @@ extern struct Queue *queue_new();
  * Destroy and free up all the data in queue
  *
  * \param[in] queue pointer to the queue, returned by queue_new
```

Comment:
```suggestion
 * \param[in,out] queue pointer to the queue, returned by queue_new
```
the `queue` variable gets modified, hence it is both an input and an output variable; please make the appropriate changes wherever valid